xgcm/test/test_padding.py

```diff
@@ -0,0 +1,1120 @@
+# TODO: Swap this out for some of our fixtures?
+import numpy as np
+import pytest
+import xarray as xr
+
+from xgcm import Grid
+from xgcm.padding import _maybe_swap_dimension_names, _strip_all_coords, pad
+from xgcm.test.datasets import datasets_grid_metric
+from xgcm.test.test_faceconnections import ds as ds_faces  # noqa: F401
+
+
+@pytest.mark.parametrize(
+    "boundary_width",
+    [
+        {"X": (1, 1)},
+        {"Y": (0, 1)},
+        {"X": (0, 1), "Y": (2, 0)},
+    ],
+)
+class TestPadding:
+    @pytest.mark.parametrize("fill_value", [np.nan, 0, 1.5])
+    def test_padding_fill(self, boundary_width, fill_value):
+
+        ds, coords, _ = datasets_grid_metric("C")
+        grid = Grid(ds, coords=coords)
+        data = ds.tracer
+
+        # iterate over all axes
+        expected = data.copy(deep=True)
+        for ax, widths in boundary_width.items():
+            dim = grid._get_dims_from_axis(data, ax)[
+                0
+            ]  # ? not entirely sure why this is a list
+            expected = expected.pad(
+                {dim: widths}, "constant", constant_values=fill_value
+            )
+
+        # we intentionally strip all coords from padded arrays
+        expected = _strip_all_coords(expected)
+
+        result = pad(
+            data,
+            grid,
+            boundary="fill",
+            boundary_width=boundary_width,
+            fill_value=fill_value,
+        )
+        xr.testing.assert_allclose(expected, result)
+
+    def test_padding_extend(self, boundary_width):
```

Comment:
I thought we were deprecating extend? Or was that another option we were deprecating?

Reply:
Oh I was thinking of extrapolate

Reply:
All in the follow up PR I would say. Ill add it to the list.

Reply:
Actually this is an interesting case. I currently have 4 tests total (fill with two values, then extend, and periodic) if I would parametrize the method aswell I would end up with 6 tests - (3 methods * 2 fill_values). In most of those cases the fill_value is not even used. Is there a nifty way to only use the fill_value parametrization when method='fill'?

Reply:
Isn't there something similar in the xhistogram test suite where it skips a bunch of parameter combinations on a particular test?